app/src/Upload.elm

```diff
@@ -55,15 +55,16 @@ type Msg
     = GotFiles (List File)
     | GotProgress Http.Progress
     | Uploaded (Result Http.Error ())
-    | NewPasswd String
+    | GotVideoNames (Result Http.Error (List String))
+    | NewPasswd Int String
```

Comment:
NewPasswd should not have to pass the "numVideos" int around. It should be matched from the existing model in update.